cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetTypeHandler.java

```diff
@@ -189,9 +209,14 @@ protected void onFinish(HttpResponder responder, File uploadedFile) throws Excep
   @Path("/data/modules/{name}")
   public void deleteModule(HttpRequest request, HttpResponder responder,
                            @PathParam("namespace-id") String namespaceId, @PathParam("name") String name) {
+    if (Constants.SYSTEM_NAMESPACE.equals(namespaceId)) {
+      responder.sendString(HttpResponseStatus.FORBIDDEN,
+                           String.format("Cannot delete modules from '%s' namespace.", namespaceId));
```

Comment:
nit: i dont like the inconsistency in errMsg for this and previous check for systemNs.